components/org.wso2.carbon.identity.oauth/src/main/java/org/wso2/carbon/identity/oauth/listener/IdentityOathEventListener.java

```diff
@@ -241,8 +241,15 @@ private boolean revokeTokensOfLockedUser(String userName, UserStoreManager userS
 
         String errorCode =
                 (String) IdentityUtil.threadLocalProperties.get().get(IdentityCoreConstants.USER_ACCOUNT_STATE);
+        if (StringUtils.isEmpty(errorCode)) {
+            errorCode =
+                    (String) IdentityUtil.threadLocalProperties.get()
+                            .get(IdentityCoreConstants.USER_ACCOUNT_STATE_WITH_USERNAME + userName);
+        }
 
         if (errorCode != null && (errorCode.equalsIgnoreCase(UserCoreConstants.ErrorCode.USER_IS_LOCKED))) {
+            IdentityUtil.threadLocalProperties.get()
```

Comment:
Where do we set this thread local?

Reply:
It is set in AccountLockHandler and AccountDisableHandler. 

Here a new identity claim is created in the AccountLockHandler https://github.com/wso2-extensions/identity-event-handler-account-lock/blob/master/components/org.wso2.carbon.identity.handler.event.account.lock/src/main/java/org/wso2/carbon/identity/handler/event/account/lock/AccountLockHandler.java#L561 during account lock flow. So this will trigger `doPreSetUserClaimValues` event again [1]and it will remove the exisiting  ` IdentityUtil.threadLocalProperties.get().remove(IdentityCoreConstants.USER_ACCOUNT_STATE);` . So the accountLock status will be removed from the threadlocal. So introduced a new threadlocal to fix this issue [2]


[1]https://github.com/wso2/carbon-identity-framework/blob/master/components/identity-mgt/org.wso2.carbon.identity.mgt/src/main/java/org/wso2/carbon/identity/mgt/IdentityMgtEventListener.java#L925
[2]https://github.com/wso2-extensions/identity-event-handler-account-lock/pull/75/files#diff-1b9c985713a862cda1296754d0c2ce127c82f08c95afc6b218e62cc16465d282R237

Reply:
https://github.com/wso2-extensions/identity-event-handler-account-lock/pull/75